.github/workflows/release-check.yml

```diff
@@ -22,8 +18,12 @@ jobs:
         with:
           go-version: "1.17.x"
       - name: Determine version
-        if: hashFiles('version.json')
-        run: echo "VERSION=$(jq -r .version version.json)" >> $GITHUB_ENV
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          version="$(gh api -X GET 'repos/${{ github.event.pull_request.head.repo.full_name }}/contents/version.json' -f ref='${{ github.event.pull_request.head.sha }}' --jq '.content' | base64 -d | jq -r '.version')"
```

Comment:
That's a very good question. In that case `version` would be an empty string and `gh: Not Found (HTTP 404)` would be printed out on stderr. The step won't fail though because the failure happens in a subshell.

I think it's non-obvious enough to warrant a comment.